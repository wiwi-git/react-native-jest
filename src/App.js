import React from 'react';
import Profile from './Profile';
import styled from 'styled-components';
import {View} from 'react-native';

const Container = styled(View)`
  flex: 1;
  align-items: center;
  justify-content: center;
`;

const App = () => {
  return (
    <Container>
      <Profile userName={'userName'} name={'name'} />
    </Container>
  );
};

export default App;
