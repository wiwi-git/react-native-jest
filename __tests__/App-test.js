/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import 'jest-styled-components';
// Note: test renderer must be required after react-native.
import {render} from '@testing-library/react-native';

let props;
let component;

function getTempComponent(props) {
  return <App {...props} />;
}

describe('[App] render', () => {
  props = {}; // fill test props
  component = getTempComponent(props);
  test('renders without crashing', () => {
    const rendered = render(component);
    expect(rendered).toMatchSnapshot(); // 기존 스냅샷과 일치하는는지.
    expect(rendered).toBeTruthy(); // 컴포넌트가 null, undefined같은 값을 가지지는 않았는지
  });
});
