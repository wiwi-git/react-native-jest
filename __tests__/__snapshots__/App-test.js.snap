// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[App] render renders without crashing 1`] = `
<View
  className=""
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      userName
      (
      name
      )
    </Text>
    <Text
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      Select your status
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Bye!
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Hello!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
